### PR TITLE
primercommit

### PR DESCRIPTION
**Pull request by smlopez88**
**Summary**:
New tests added for the CF01 Sign Up User Story

- empty-fields-errors.js: a test for each field (username, email, password)
- existing-user.js: three tests (repeated email and username, repeated email, repeated username)
- password-validations.js: three tests (password with less than 8 characters, password with blank spaces, deleted password)
- email-validations.js: two tests (existing email, email with no @ )
- successful-signup.js one test (happy signup path)

**Screenshots:**

PR

cypress

### CODE REVIEW

File: CF01/email-validations.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+const { expect } = require("chai");
+const { beforeEach } = require("mocha");
+
+describe ('Email Errors', function(){
+    beforeEach(function() {
+        cy.visit('https://qa-angular-conduit.vercel.app/register');
```
Comment (eledwinn) [inline CF01/email-validations.js]: Add base url in cypress.json file.
https://docs.cypress.io/guides/references/configuration.html#Options
And in this line, only use `cy.visit('register');`

File: CF01/email-validations.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+const { expect } = require("chai");
+const { beforeEach } = require("mocha");
+
+describe ('Email Errors', function(){
+    beforeEach(function() {
+        cy.visit('https://qa-angular-conduit.vercel.app/register'); 
+    })
+    afterEach(function() {
+        cy.wait(6000);
```
Comment (eledwinn) [inline CF01/email-validations.js]: Why need this wait after each?

File: CF01/email-validations.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+const { expect } = require("chai");
```
Comment (eledwinn) [inline CF01/email-validations.js:1]: Use two spaces as indent configuration please

File: CF01/email-validations.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        cy.wait(6000);
+    })
+     
+    it('Existing Email',function(){
+        cy.get(':nth-child(1) > .form-control').type('sml0pez88');
+        cy.get(':nth-child(2) > .form-control').type('santiago.lopez@wolox.com.ar');
+        cy.get(':nth-child(3) > .form-control').type('Pass1234');
+        cy.get('.btn').click();
+        cy.get('.error-messages > :nth-child(1)').should('contain', 'email is already taken');        
+        })
```
Comment (eledwinn) [inline CF01/email-validations.js]: Indent this line please

File: CF01/email-validations.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+describe ('Email Errors', function(){
+    beforeEach(function() {
+        cy.visit('https://qa-angular-conduit.vercel.app/register'); 
+    })
+    afterEach(function() {
+        cy.wait(6000);
+    })
+     
+    it('Existing Email',function(){
```
Comment (eledwinn) [inline CF01/email-validations.js]: Add white space after comma, and after `()` 😃

File: CF01/email-validations.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        cy.get(':nth-child(1) > .form-control').type('sml0pez88');
```
Comment (eledwinn) [inline CF01/email-validations.js]: Try find a better selector, like id, name or another attribute. (only if possible)

File: CF01/email-validations.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        })
+
+    it ('No @',function(){
+        cy.get(':nth-child(1) > .form-control').type('sml0pez88');
+        cy.get(':nth-child(2) > .form-control').type('santiago.lopezwolox.com.ar');
+        cy.get(':nth-child(3) > .form-control').type('Pass1234');
+        cy.get('.btn').click();
+        cy.get('.error-messages > :nth-child(1)').should('contain', 'email is invalid');        
+        })
+
```
Comment (eledwinn) [inline CF01/email-validations.js]: Remove empty line

File: CF01/email-validations.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+     
+    it('Existing Email',function(){
+        cy.get(':nth-child(1) > .form-control').type('sml0pez88');
+        cy.get(':nth-child(2) > .form-control').type('santiago.lopez@wolox.com.ar');
+        cy.get(':nth-child(3) > .form-control').type('Pass1234');
+        cy.get('.btn').click();
+        cy.get('.error-messages > :nth-child(1)').should('contain', 'email is already taken');        
+        })
+
+    it ('No @',function(){
```
Comment (eledwinn) [inline CF01/email-validations.js]: Add white space after comma and after `()`

File: CF01/email-validations.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+        cy.get(':nth-child(2) > .form-control').type('santiago.lopezwolox.com.ar');
+        cy.get(':nth-child(3) > .form-control').type('Pass1234');
+        cy.get('.btn').click();
+        cy.get('.error-messages > :nth-child(1)').should('contain', 'email is invalid');        
+        })
+        
+    })
+
+
+
```
Comment (eledwinn) [inline CF01/email-validations.js]: Only one empty line please

File: CF01/empty-fields-error.js [hunk 10]
```diff
@@ -0,0 +1,49 @@
+const { beforeEach } = require("mocha");
+
+describe ('Empty Fields', function(){
+    beforeEach(function() {
+        cy.visit('https://qa-angular-conduit.vercel.app/register');
```
Comment (eledwinn) [inline CF01/empty-fields-error.js]: Use baseUrl :)

File: CF01/empty-fields-error.js [hunk 11]
```diff
@@ -0,0 +1,49 @@
+const { beforeEach } = require("mocha");
+
+describe ('Empty Fields', function(){
+    beforeEach(function() {
+        cy.visit('https://qa-angular-conduit.vercel.app/register'); 
+    })
+    afterEach(function() {
```
Comment (eledwinn) [inline CF01/empty-fields-error.js]: Empty line between each function

File: CF01/empty-fields-error.js [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+        cy.wait(1000);
+    })
+       
+    it('Empty Password',function(){
+        cy.get(':nth-child(1) > .form-control').type('santiago.lopez');
+        cy.get(':nth-child(2) > .form-control').type('santiago.lopez@wolax.com.ar');
+        cy.get('.btn').then((btn) => {
+            if (cy.get('.btn').should('have.attr','disabled')==false) {
```
Comment (eledwinn) [inline CF01/empty-fields-error.js]: did you test this? I think the .should method doesn't work as you expect. 
try to test for both cases to make sure it works please.

File: CF01/empty-fields-error.js [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+                cy.get('.btn').click()
+            } else {
+                cy.get(':nth-child(1) > .nav-link').click()
+            }
+
```
Comment (eledwinn) [inline CF01/empty-fields-error.js]: Remove empty lines between internal blocks

File: CF01/empty-fields-error.js [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+        cy.get('.btn').then((btn) => {
+            if (cy.get('.btn').should('not.have.attr','disabled')) {
+                cy.get('.btn').click()
+            } else {
+                cy.get(':nth-child(1) > .nav-link').click()
+            }
+        })
+    })
+
+})
```
Comment (eledwinn) [inline CF01/empty-fields-error.js]: Always add empty line at end of files

File: CF01/password-validations.js [hunk 15]
```diff
@@ -0,0 +1,33 @@
+const { expect } = require("chai");
+const { beforeEach } = require("mocha");
+
+describe ('Password Errors', function(){
+    beforeEach(function() {
+        cy.visit('https://qa-angular-conduit.vercel.app/register'); 
+    })
+    afterEach(function() {
```
Comment (eledwinn) [inline CF01/password-validations.js]: Empty line between functions

File: CF01/successful-signup.js [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+    afterEach(function() {
+        cy.wait(6000);
+    })
+     
+    it ('Sucessful Sign Up',function(){
+        cy.get(':nth-child(1) > .form-control').type('TestUser2002');
+        cy.get(':nth-child(2) > .form-control').type('santiago.lopez+234@wolox.com.ar');
+        cy.get(':nth-child(3) > .form-control').type('Wolox1189!');
+        cy.get('.btn').click();  
+        cy.wait(6000);
```
Comment (eledwinn) [inline CF01/successful-signup.js]: Avoid use `cy.wait`, if you need wait for next command, use selector timeout.
```js
cy.get(':nth-child(4) > .nav-link', { timeout: 6000 })
```

**Review by eledwinn — CHANGES_REQUESTED** [review]

**Comment by smlopez88** [comment]
All changes added in second commit.

**Pull request merged**